src/Uno.UWP/ApplicationModel/Contacts/ContactReader.Android.cs

```diff
@@ -0,0 +1,370 @@
+﻿
+#if __ANDROID__
+using System;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+using Windows.Foundation;
+
+namespace Windows.ApplicationModel.Contacts
+{
+	public partial class ContactReader
+	{
+		internal ContactQueryOptions _queryOptions;
+		private Android.Database.ICursor _cursor = null;
+		private Android.Content.ContentResolver _contentResolver = null;
+
+		internal ContactReader(ContactQueryOptions options)
+		{
+			if (options is null)
+				throw new ArgumentNullException();
+
+			_queryOptions = options;
+
+			Android.Net.Uri oUri;
+			string sColumnIdName = "_id";
+
+			switch (options.SearchFields)
+			{
+				case ContactQuerySearchFields.Phone:
+					oUri = Android.Net.Uri.WithAppendedPath(
+						Android.Provider.ContactsContract.PhoneLookup.ContentFilterUri, // Phone.Contact_ID mapped to .Contacts._ID 
+						Android.Net.Uri.Encode(options.SearchText));
+					sColumnIdName = "contact_id";
+					break;
+				case ContactQuerySearchFields.Name:
+					oUri = Android.Net.Uri.WithAppendedPath(
+						Android.Provider.ContactsContract.Contacts.ContentFilterUri,
+						Android.Net.Uri.Encode(options.SearchText));
+					break;
+				default:
+					oUri = Android.Provider.ContactsContract.Contacts.ContentUri; 
+					break;
+			}
+
+
+			_contentResolver = Android.App.Application.Context.ContentResolver;
+
+			_cursor = _contentResolver.Query(oUri,
+									new string[] { sColumnIdName, "display_name" },  // which columns
+									null,   // where
+									null,   
+									null);   // == date DESC
+			if(!_cursor.MoveToFirst())
+			{
+				_cursor.Dispose();
+				_cursor = null;
+			}
+		}
+
+		~ContactReader()
+		{
+			if(_cursor != null)
+			{
+				_cursor.Dispose();
+				_cursor = null;
+			}
+		}
+
+
+		private async Task<ContactBatch> ReadBatchAsyncTask()
+		{
+			ContactBatch batch = new ContactBatch(ReadBatchInternal());
+			return batch;
+		}
+
+		public IAsyncOperation<ContactBatch> ReadBatchAsync()
+			=> ReadBatchAsyncTask().AsAsyncOperation();
+
+		internal List<Contact> ReadBatchInternal()

```

Comment:
Why does it need to be internal ?

Reply:
Good question. Answer: I don't remember. This PR is from January 2020. Trying to change it to private.